local-volume/provisioner/pkg/discovery/discovery.go

```diff
@@ -207,21 +203,21 @@ func generatePVName(file, node, class string) string {
 	return fmt.Sprintf("local-pv-%x", h.Sum32())
 }
 
-func (d *Discoverer) createPV(file, class string, config common.MountConfig, capacityByte int64, volType string) {
+func (d *Discoverer) createPV(file, class string, config common.MountConfig, capacityByte int64, volMode v1.PersistentVolumeMode) {
 	pvName := generatePVName(file, d.Node.Name, class)
 	outsidePath := filepath.Join(config.HostDir, file)
 
-	glog.Infof("Found new volume of volumeType %q at host path %q with capacity %d, creating Local PV %q",
-		volType, outsidePath, capacityByte, pvName)
+	glog.Infof("Found new volume of volumeMode %q at host path %q with capacity %d, creating Local PV %q",
+		volMode, outsidePath, capacityByte, pvName)
 
-	// TODO: Set block volumeType when the API is ready.
 	pvSpec := common.CreateLocalPVSpec(&common.LocalPVConfig{
```

Comment:
Do you want to consider checking the volume mode on the PV after creation?  And if the volume mode doesn't match, then log a warning message?

Reply:
I have added log message which calls out any disconnect between discovered volume mode and PV's volume mode.